packages/typescript-estree/src/parseSettings/createParseSettings.ts

```diff
@@ -80,10 +80,10 @@ export function createParseSettings(
     errorOnTypeScriptSyntacticAndSemanticIssues: false,
     errorOnUnknownASTType: options.errorOnUnknownASTType === true,
     EXPERIMENTAL_projectService:
-      (options.EXPERIMENTAL_useProjectService &&
-        process.env.TYPESCRIPT_ESLINT_EXPERIMENTAL_TSSERVER !== 'false') ||
-      (process.env.TYPESCRIPT_ESLINT_EXPERIMENTAL_TSSERVER === 'true' &&
-        options.EXPERIMENTAL_useProjectService !== false)
+      options.EXPERIMENTAL_useProjectService ||
+      (options.project &&
```

Comment:
Hello @JoshuaKGoldberg , should we check it's non-empty?

I don't know this `project` options, asking this only because I use `project:[]` in [prettier](https://github.com/prettier/prettier/blob/9db174aadd612ebade27bf8c00ba83dbbc68117c/src/language-js/parse/typescript.js#L20) and [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/41ac4228c6edbbcb7c29bd568b7695e36755cb4d/test/utils/parsers.mjs#L51), I don't remember why, maybe prevent search for `tsconfig.json`?

Reply:
Looking at [the docs](https://typescript-eslint.io/packages/parser#project), maybe I should use `project: false`?

Reply:
Hmm, yes, or `project: null`.

Reply:
Back to my question, shouldwe treat `[]` as false?

Reply:
Good question 🤔. My gut instinct is that it should probably be an error. I'm not sure though... Filed #8210.